tests/MongoDB.Driver.Core.Tests/Core/Misc/ServerVersionTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using FluentAssertions;
+using Xunit;
+
+namespace MongoDB.Driver.Core.Misc
+{
+    public class ServerVersionTests
+    {
+        [Theory]
+        [InlineData(null, null, 0)]
+        [InlineData(null, "1.0.0", -1)]
+        [InlineData("1.0.0", null, 1)]
+        [InlineData("1.0.0", "1.0.0", 0)]
+        [InlineData("1.1.0", "1.1.0", 0)]
+        [InlineData("1.1.1", "1.1.1", 0)]
+        [InlineData("1.0.0", "2.0.0", -1)]
+        [InlineData("2.0.0", "1.0.0", 1)]
+        [InlineData("1.0.0", "1.1.0", -1)]
+        [InlineData("1.1.0", "1.0.0", 1)]
+        [InlineData("1.0.0", "1.0.1", -1)]
+        [InlineData("1.0.1", "1.0.0", 1)]
+        [InlineData("4.4.0-rc12", "4.4.0-rc12", 0)]
+        [InlineData("4.4.0-rc12", "4.4.0", -1)]
+        [InlineData("4.4.1-rc12", "4.4.0", 1)]
+        [InlineData("4.4.0-rc13", "4.4.0-rc12", 1)]
+        [InlineData("4.5.0-489-gb8f58d7", "4.5.0-489-gb8f58d7", 0)]
+        [InlineData("4.5.0-489-gb8f58d7", "4.5.1", -1)]
+        [InlineData("4.5.0-489-gb8f58d7", "4.5.0", 1)]
+        [InlineData("4.5.0-489-gb8f58d7", "4.5.0-5-g5a9a742f6f", 1)]
+        [InlineData("4.4.0-rc12-5-g5a9a742f6f", "4.4.0-rc12-5-g5a9a742f6f", 0)]
+        [InlineData("4.4.0-rc12-5-g5a9a742f6f", "4.4.0-rc13", -1)]
+        [InlineData("4.4.0-rc12-5-g5a9a742f6f", "4.4.0-rc12", 1)]
+        [InlineData("4.4.0-rc12-5-g5a9a742f6f", "4.4.0-rc12-4-g5a9a742f6f", 1)]
+        [InlineData("4.4.0-rc12-5-g5a9a742f6f", "4.4.0-rc13-5-g5a9a742f6f", -1)]
+        public void Comparisons_should_be_correct(string a, string b, int comparison)
+        {
+            var subject = a == null ? null : ServerVersion.Parse(a);
+            var comparand = b == null ? null : ServerVersion.Parse(b);
+            subject?.Equals(comparand).Should().Be(comparison == 0);
+            subject?.CompareTo(comparand).Should().Be(comparison);
+            (subject == comparand).Should().Be(comparison == 0);
```

Comment:
nit: I have no idea what happens here..

Reply:
Testing all the comparison methods and overloads in one place.

Reply:
my point was that it's unclear from the code reading point of view what happens here. But since I marked my comment as `nit`. Then it's ok

Reply:
It was already this way in `SemanticVersionTests.cs` and Mikalai just left the test the same as it was originally.

Reply:
In short, this code allows to see inconsistencies in operator overloads